hw/formal/README.md

```diff
@@ -311,6 +311,19 @@ For example:
           $past(rv_plic.le[src_sel]) || $past(intr_src_i[src_sel]), clk_i, !rst_ni)
 ```
 
+## Implementation Guidelines
+Assertions are widely used in RTL and FPV files. Below decumented the
+recommended guidelines in terms of where to implement assertions.
+* Basic assertions are recommended to implement directly under the RTL file.
+  This is commonly used for designers to write basic functional assertions for
+  sanity check
+* Assertions used for testbench to achieve verification goals are recommended to
+  implement under the `ip/hw/module_name/fpv/vip` folder. This FPV environment can be
+  automatically generate by the [python script](../../util/fpvgen/README.md)
+* Assertions shared among the same interfaces/modules are recommended to
```

Comment:
"Assertions shared among the same interface/modules should be implemented in a common file and bound to the testbench. ..."?

This bulletpoint should also indicate explicitly where that common file should be placed.

Reply:
Thanks Alex for your detailed suggestions! Much appreciated and updated all of them. I also rephrase the last bullet-point, as it was not really clear..